### ISSUE TITLE
Features omitted from rules-based symbology should not be output

### ISSUE THREAD

**Issue by bkepl** [qgis/QGIS#31862]
I use rules-based symbology quite a lot; it saves me from duplicating a layer several times and querying the layers to achieve my desired effects. When using a rule-based symbology for a parcels layer to display only a few parcels, the PDF created is huge because the parcels omitted by my rule were nonetheless created in the exported PDF, but with no stroke or fill. These features should not have been created in the exported PDF, just as if I wrote a query on the layer.

**Comment by bkepl** [comment]
I need to qualify this a little. The problem I refer to above occurs when I nest symbologies. I set the fill and outline of the parent to transparent while the children have visible symbologies. I find it very useful to group complex symbologies in this manner (for this to work, symbologies are necessarily rules-based). The problem is that setting fill and outline to transparent on the parent means only that the features will be transparent, but they will still be output, so a resulting PDF can be quite large. 

I will do this sort of grouping for labeling, which for me is a very intuitive way of keeping various labeling styles for differing purposes organized. I recently noticed that labeling for a rule can be set to “No labeling,” which is how I will set the parent of the group. It would be very useful if if a rules-based symbology could similarly set to “No symbology,” just as a layer can be set to “No symbology.”

**Comment by bkepl** [comment]
As I was working more with this problem, I discovered a couple things:

1. When I group symbologies, a filter at the parent level controls the display of features in child symbologies. 
2. One can, in fact, suppress the output of features for the parent symbology. I found this by accident: down toward the bottom of the style panel, there is a checkbox that, when checked, suppresses rendering of the features, but only at that level; child symbologies will still display.
